### PR TITLE
Remove KeepAllProjectReferences=true from test projects

### PR DESCRIPTION
**Pull request by sepidehMS**
Addresses https://github.com/dotnet/corefx/issues/7913
Building some of the test projects with `/p:BuildTestsAgainstPackages=true` threw errors and that's why `KeepAllProjectReferences=true` was added to keep the project references.

This PR removes this property to make sure we can successfully build and run those tests with only referencing nupkg.

**Note:** I wasn't able to remove that from the following as these were testing Internal classes. Any ideas on what should be done about these?

```
System.Diagnostics.Debug.Tests
System.Linq.Expressions.Tests
```

cc: @weshaggard @danmosemsft @stephentoub @joperezr @chcosta

### CODE REVIEW

File: src/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/project.json [hunk 1]
```diff
@@ -19,6 +19,8 @@
     "System.Threading": "4.4.0-beta-24619-02",
     "System.Threading.Thread": "4.4.0-beta-24619-02",
     "System.Threading.Tasks": "4.4.0-beta-24619-02",
+    "System.Xml.ReaderWriter": "4.4.0-beta-24619-02",
+    "System.Xml.XPath": "4.4.0-beta-24619-02",
```
Comment (sepidehMS) [inline src/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/project.json]: The reason I did not remove this project.json is in the [common tests' project.json](https://github.com/dotnet/corefx/blob/master/src/Common/test-runtime/project.json) `System.Runtime.Loader` does not have a run-time assembly for net463 on some platforms (win-x86, win-x64) supported there.

**Comment by weshaggard** [comment]
>Note: I wasn't able to remove that from the following as these were testing Internal classes. Any ideas on what should be done about these?

For now just leave them, the fact that they depend on internals is going to cause those test to be non-portable and we will have to build them for each of the concrete platforms to run the same tests.

**Comment by weshaggard** [comment]
With these changes have you verified that they build and run correctly against packages?

**Comment by sepidehMS** [comment]
> With these changes have you verified that they build and run correctly against packages?

@weshaggard Yes, I was able to successfully run the tests following the steps mentioned [here](https://github.com/dotnet/corefx/blob/master/Documentation/building/build-tests-against-packages.md)

File: src/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/project.json (hunk 1 above)
Comment (weshaggard) [inline src/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/project.json]: If these tests depend on System.Runtime.Loader then they should be converted to be netcoreapp1.1 specific and then you should be able to remove this project.json.

**Comment by weshaggard** [comment]
Assuming these will work in our official builds when running against packages. The changes look good to me.

**Comment by sepidehMS** [comment]
@weshaggard This test project already has `<NuGetTargetMoniker>.NETStandard,Version=v1.7</NuGetTargetMoniker>`, but the problem is when I tried to remove the project.json file and use the common test project.json instead, at run-time one test failed with `System.IO.FileNotFoundException: Could not load "System.Runtime.Loader" ...` and that is becuase in the project.json we have 
```
"System.Runtime.Loader": {
          "version": "4.4.0-beta-24619-02",
          "include": "compile"
        },
```
and if you remove `"include": "compile"` here, the build from root fails with:
`System.Runtime.Loader 4.4.0-beta-24619-02 provides a compile-time reference assembly for System.Runtime.Loader on .NETFramework,Version=v4.6.3, but there is no run-time assembly compatible with win-x64.` which should be the reason why that was added in the project.json in the first place.

**Comment by weshaggard** [comment]
You need to build the test project directly against netcoreapp1.1, not netstandard1.7, in order to consume System.Runtime.Loader. The other option is to remove the net463 TestTFM for this project because net463 doesn't support System.Runtime.Loader.

**Comment by sepidehMS** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test please
@dotnet-bot test  Innerloop Linux ARM Emulator Release Cross Build please

**Pull request merged**